src/prefect/orion/schemas/actions.py

```diff
@@ -196,6 +229,38 @@ def remove_old_fields(cls, values):
     storage_document_id: Optional[UUID] = FieldFrom(schemas.core.Deployment)
     infrastructure_document_id: Optional[UUID] = FieldFrom(schemas.core.Deployment)
 
+    @staticmethod
+    def _get_base_config_defaults(variables: dict) -> dict:
+        """Get default values from base config for all variables that have them."""
+        defaults = dict()
+        for variable_name, attrs in variables.items():
+            if "default" in attrs:
+                defaults[variable_name] = attrs["default"]
+
+        return defaults
+
+    @staticmethod
+    def _validate_variables(variables_schema: dict, variables: dict):
+        """Check that variables conform to specified constraints."""
+        schema = {"type": "object", "properties": variables_schema}
+        jsonschema.validate(variables, schema)
+
+    def check_valid_configuration(self, base_job_template: dict):
+        """Check that configuration has all required variables"""
+        variables_schema = base_job_template.get("variables")
+        required_variables = base_job_template.get("required")
+
+        missing_variables = set(required_variables).difference(self.infra_overrides)
```

Comment:
Let jsonschema handle this

Reply:
If you include the required array with the variable schema then I think this could be reduced to a single `jsonschema.validate` call. That would solve your duplicated code problem.